contentcuration/contentcuration/frontend/shared/data/resources.js

```diff
@@ -1416,6 +1416,36 @@ export const ContentNode = new TreeResource({
     });
   },
   setChannelIdOnChange: setChannelIdFromTransactionSource,
+
+  /**
+   * Waits for copying of content nodes to complete for all ids referenced in `ids` array
+   * @param {string[]} ids
+   * @returns {Promise<void>}
+   */
+  waitForCopying(ids) {
+    const observable = Dexie.liveQuery(() => {
+      return this.table
+        .where('id')
+        .anyOf(ids)
+        .filter(node => !node[COPYING_FLAG])
+        .toArray();
+    });
+
+    return new Promise((resolve, reject) => {
```

Comment:
I think your comment got lost in a rebase, but this does look like a neat way to watch for changes.